helios-client/src/main/java/com/spotify/helios/common/descriptors/RolloutOptions.java

```diff
@@ -79,31 +92,48 @@ public Builder toBuilder() {
         .setTimeout(timeout)
         .setParallelism(parallelism)
         .setMigrate(migrate)
+        .setOverlap(overlap)
         .setToken(token)
         .setIgnoreFailures(ignoreFailures);
   }
 
-  public long getTimeout() {
+
+  /**
+   * Return a new RolloutOptions instance by merging this instance with another one.
+   * The other instance is assumed to have no null-valued fields.
```

Comment:
what happens if this assumption is violated?

Reply:
`firstNonNull()` will throw NPE if both are null. So I think this is OK.